File: backend/src/main/java/com/igreja/site/model/Pregacao.java
package com.igreja.site.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;

@Entity
@Table(name = "pregacoes")
public class Pregacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String titulo;

    @NotNull
    private LocalDate data;

    private String tema; // ex: Fe, Familia, Esperanca

    private String videoUrl; // link do video (Youtube, etc)

    @Column(length = 3000)
    private String descricao;

    @ManyToOne
    @JoinColumn(name = "pregador_id")
    @JsonIgnoreProperties({"biografia"})
    private Pregador pregador;

    public Pregacao() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public LocalDate getData() { return data; }
    public void setData(LocalDate data) { this.data = data; }

    public String getTema() { return tema; }
    public void setTema(String tema) { this.tema = tema; }

    public String getVideoUrl() { return videoUrl; }
    public void setVideoUrl(String videoUrl) { this.videoUrl = videoUrl; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }

    public Pregador getPregador() { return pregador; }
    public void setPregador(Pregador pregador) { this.pregador = pregador; }
}

File: backend/src/models/Pregacao.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

// Equivalente a Pregacao.java
const Pregacao = sequelize.define(
  'Pregacao',
  {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    titulo: { type: DataTypes.STRING, allowNull: false, validate: { notEmpty: true } },
    data: { type: DataTypes.DATEONLY, allowNull: false },
    // ex: Fe, Familia, Esperanca
    tema: { type: DataTypes.STRING, allowNull: true },
    // link do video (Youtube, etc)
    videoUrl: { type: DataTypes.STRING, allowNull: true },
    descricao: { type: DataTypes.STRING(3000), allowNull: true },
    pregadorId: { type: DataTypes.INTEGER, allowNull: true, field: 'pregador_id' },
  },
  {
    tableName: 'pregacoes',
    timestamps: false,
  }
);

module.exports = Pregacao;
